tls/s2n_connection.c

```diff
@@ -977,9 +977,29 @@ const char *s2n_connection_get_cipher(struct s2n_connection *conn)
     notnull_check_ptr(conn);
     notnull_check_ptr(conn->secure.cipher_suite);
 
+    /* ensure we've negotiated a cipher suite */
+    ENSURE_PTR(memcmp(conn->secure.cipher_suite->iana_value, s2n_null_cipher_suite.iana_value, sizeof(s2n_null_cipher_suite.iana_value)) != 0, S2N_ERR_INVALID_STATE);
```

Comment:
Much as I'd love to return a descriptive error here, I don't think we can safely change this behavior. Even though people probably only use this for debugging, it's still a public API.

Reply:
Wouldn't it be better to use `EXPECT_EQ(memcmp(..), 0);`  instead here? 

Or  if we want to keep the ENSURE macros use `ENSURE_EQ(memcmp(conn->secure.cipher_suite->iana_value, s2n_null_cipher_suite.iana_value, sizeof(s2n_null_cipher_suite.iana_value)), 0);`  This way the error message would be `S2N_ERR_SAFETY`.

Reply:
> Wouldn't it be better to use EXPECT_EQ(memcmp(..), 0); instead here?

This isn't a test so EXPECT_EQ isn't available

> This way the error message would be S2N_ERR_SAFETY.

I want the `S2N_ERR_INVALID_STATE` so it's more helpful to the caller